apps/base-docs/src/components/DocChat/ChatModal.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useState, useCallback, useEffect, useRef } from 'react';
+import Modal from '../Modal';
+import ChatMessage from './ChatMessage';
+import Icon from '../Icon';
+
+import styles from './styles.module.css';
+
+import {
+  ConversationMessage,
+  ChatHistoryMessage,
+  getConversationId,
+  setSessionConversation,
+  getSessionConversation,
+  streamPromptResponse,
+  controller,
+} from './docChat';
+
+type ChatModalProps = {
+  visible: boolean;
+  onRequestClose: () => void;
+};
+
+export default function ChatModal({ visible, onRequestClose }: ChatModalProps) {
+  const [conversationId, setConversationId] = useState(0);
+  const [conversation, setConversation] = useState<ConversationMessage[]>([]);
+  const [chatHistory, setChatHistory] = useState<ChatHistoryMessage[]>([]);
+  const [prompt, setPrompt] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [isGenerating, setIsGenerating] = useState(false);
+
+  const conversationContainerRef = useRef<HTMLDivElement>(null);
+  const currentMessage: ConversationMessage = conversation[conversation.length - 1];
+
+  const handleInputChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => setPrompt(e.target.value),
+    [prompt],
+  );
+
+  const handleSubmit = useCallback(
+    (e: React.SyntheticEvent) => {
+      e.preventDefault();
+
+      if (!conversationId || !prompt || isLoading || isGenerating) return;
+
+      setIsLoading(true);
+
+      setChatHistory((prevState: ChatHistoryMessage[]) => {
+        const newState = [...prevState];
+        newState.push({ prompt, response: '' });
+        return newState;
```

Comment:
Can write this more simply as `return [...prevState, { prompt, response: ''}]`

Reply:
Fixed 👍